### ISSUE TITLE
cachet.components.status.0 upon updating Component from API

### ISSUE THREAD

**Issue by nadeaus** [CachetHQ/Cachet#2854]
After digging around the internet for this known issue - I was unable to find a solution that works for us.

I'm trying to do a simple CURL request to update the status of a certain component. Here is the command:

curl -H "Content-Type: application/json;" -H "X-Cachet-Token: xxxxxxxxxxx" --request PUT --url http://status.xxxxxx.com/api/v1/components/29 --data '{"status":2}'

Each time I run this command, the component status gets set to 0, and looks like this:

cachet.components.status.0

Can someone please help me shine some light on this issue?

**Comment by aojari** [comment]
I am having the same issue but in addition am getting problems creating incidents.  I get a message back saying that name field is required etc despite the fact that I have defined them.

**Comment by jbrooksuk** [comment]
What version of Cachet are you using?

**Comment by aojari** [comment]
I have tried it on 2.4 and 2.3.13 both with the same result

**Comment by 2Belette** [comment]
hi,
I am struggling after multiple hours of testing with no result and I am wondering if I am not on the same situation.
I am trying to simply update a component with a PUT request.
My idea is to simply update the status of the component (but I also tested with other fields) and no success but no error...nor authentication issue nothing only a 200 OK :( :)

Using requests from python & Cachet 2.4.0-dev

Not sure it is related to this issue.
If you have any idea! will spend more hours on it anyway...

**Comment by nadeaus** [comment]
Currently on version 2.3.9 of Cachet @jbrooksuk

**Comment by 2Belette** [comment]
@jbrooksuk : any idea / tips regarding how to update a component status? Is it possible to do it like it is documented ? (see my previous comment here).
I tried hard but I am unable to make it work :(

**Comment by finndev** [comment]
Same issue, latest version of cachet. Unable to update component's status using a PUT request as documented per API.

**Comment by nadeaus** [comment]
I'm on version 2.3.9. Any ideas?

**Comment by tomaik** [comment]
@nadeaus @aojari @finndev  @2Belette 

I'm using 2.3.13 version (docker image) and it works for me.

`curl -X PUT \
  http://example.com/api/v1/components/25 \
  -H 'Content-Type: application/x-www-form-urlencoded' \
  -H 'X-Cachet-Token: xxxxxxxxxxx' \
  -d status=1`

**Comment by marcel-dempers** [comment]
Having the same issue even with 2.3.13 version of the docker image. 
I can confirm the `curl` command as per @tomaik works. However using POST via JSON object does not work.

Running a PUT from postman 
http://localhost:8000/api/v1/components/1

JSON:
```
{
    "data": {
        "status": 1
    }
}
```
This gets a success response but when I do a GET to see the latest object I get these fields:

```
{
    "data": {
        "id": 1,
        "status": 0,
        "status_name": "cachet.components.status.0"
       
    }
}

```
So PUT using JSON as input does not appear to be working correctly

**Comment by marcel-dempers** [comment]
Hold on a sec. I think I found it. When I crafted my call to API server, the API documentation somehow led me to believe that I needed a body like this:

```
{
    "data": {
        "status": 1
    }
}
```

However, I needed to omit the `data` part, so updated my PUT request body to 
```
{
    
    "status": 3
  
}
```
And then it worked correctly.

User issue on my part lol, but I hope this helps someone just in case

**Comment by jbrooksuk** [comment]
@marcel-dempers where do you find that in the documentation? We can make it more obvious.

**Comment by marcel-dempers** [comment]
I think its more of a user interpretation \ UX thing @ https://docs.cachethq.io/reference#update-a-component

The way I interpreted it was by reading it from top (left to right ) and then bottom, I saw the top left as the request (which is correct) and I interpreted the right as the input parameters (not the expected response). When I read downwards, it specified the fields. Therefore I was thinking all the fields below needed to be wrapped in a form to match the right hand pane, so I mistakenly wrapped my params in a `data : {}` object.

Hope that makes sense :) It might just be me
